boot/zephyr/flash_map_extended.c

```diff
@@ -20,7 +20,23 @@
 
 BOOT_LOG_MODULE_DECLARE(mcuboot);
 
-#if (!defined(CONFIG_XTENSA) && DT_HAS_CHOSEN(zephyr_flash_controller))
+#if defined(CONFIG_STM32_MEMMAP)
+/* MEMORY MAPPED for XiP on external NOR flash takes the sspi-nor or ospi-nor or qspi-nor device */
+#define FLASH_DEVICE_ID SPI_FLASH_0_ID
+#if DT_NODE_HAS_STATUS(DT_INST(0, st_stm32_xspi_nor), okay)
+#define FLASH_DEVICE_NODE DT_INST(0, st_stm32_xspi_nor)
+#define FLASH_DEVICE_BASE DT_REG_ADDR_BY_IDX(DT_INST(0, st_stm32_xspi),1)
```

Comment:
Instance 0 of `st_stm32_xspi` could be used for other purposes (PSRAM on N6-DK for instance).
We should ensure we get the parent of the `st_stm32_xspi_nor` node instead.
Something like:
```suggestion
#define DT_DRV_COMPAT st_stm32_xspi_nor
#define FLASH_DEVICE_BASE DT_REG_ADDR_BY_IDX(DT_INST_PARENT(0), 1)
```

Reply:
I just ran into this exact issue on my STM32H7S78-DK, see https://github.com/zephyrproject-rtos/zephyr/pull/88052#issuecomment-2990216467

This is because the flash for XiP is on XSPI2, and there is a PSRAM on XSPI1 which is therefore enabled in the device tree. Consequently, the macro being reviewed here is mistakenly using the XSPI1 (PSRAM) address space instead of XSPI2.

I solved the issue with the following change:
```suggestion
#define FLASH_DEVICE_BASE DT_REG_ADDR_BY_IDX(DT_PARENT(FLASH_DEVICE_NODE),1)
```

I think we could do the same for ospi/qspi as well